10a-walkthrough-6.Rmd

```diff
@@ -0,0 +1,677 @@
+
+```{r plot without outliers}
+child_counts %>% 
+    filter(age == "Total, Age 6-21") %>%
+    spread(gender, total) %>% 
+    # Filter for female student counts less than 500,000
+    filter(f <= 500000) %>%
+    ggplot(aes(x = f, y = m)) + 
+    geom_point(size = 3, alpha = .5) + 
+    labs(title = "Comparison of female students to male students in special education", 
+         subtitle = "Counts of students in each state, ages 6-21.\nDoes not include outlying areas and freely associated states", 
+         x = "Female students", 
+         y = "Male students", 
+         caption = "Data: US Dept of Education")
+```
+
+Model differences in identification by gender: 
+
+```{r make model data}
+model_data <- child_counts %>% 
+    filter(age == "Total, Age 6-21") %>% 
+    mutate(year = as.factor(year(year))) %>% 
+    spread(gender, total) %>% 
+    # Exclude outliers
+    filter(f < 500000) %>% 
+    # Compute male student to female student ratio 
+    mutate(ratio = m / f) %>% 
+    select(-c(age, disability))
+```
+
+```{r check factor sample sizes}
+model_data %>% count(year)
+```
+
+Earlier we asked the question Do we have enough data points for the count of female students to learn about the ratio of female to male students? Similarly, we should ask the question do we have enough data points across our year variable to learn about how the ratio of female to male students has changed over time. We can visualize the ratio values across all years to check. Note the use of `geom_jitter` to spread the points horizontally so we can estimate the quantities better: 
+
+```{r}
+ggplot(data = model_data, aes(x = year, y = ratio)) + 
+    geom_jitter(alpha = .5) + 
+    labs(title = "Male to female ratio across years (jittered)")
+```
+
+Each year seems to have data points that can be considered when we fit the model. This means that there are enough data points to help us learn how the year variable predicts the ratio variable. 
+
+We fit the linear regression model by passing the argument `ratio ~ year` to the function `lm`. In R, the `~` usually indicates a formula. In this case, the formula is the variable `year` as a predictor of the variable `ratio`. The final argument we pass to `lm` is `data = model_data`, which tells R to look for the variables `ratio` and `year` in the dataset `model_data`. The results of the model are called a "model object." We'll store the model object in `ratio_year`:
+
+```{r fit model for year predicting ratio }
+ratio_year <- lm(ratio ~ year, data = model_data)
```

Comment:
I just really love this analysis.

Reply:
Something I'm wondering about mentioning here: can we refer back to another section on multi-level models - thinking of mentioning that it could be possible to nest the data within states (and to obtain state-specific estimates), but that that is not our focus here.